apps/o2-blocks/src/callout/editor.js

```diff
@@ -0,0 +1,331 @@
+			if ( iconName !== customIcon ) {
+				setAttributes( { customIcon: iconName } );
+			}
+		},
+		[ customIcon, setAttributes ]
+	);
+
+	const handleHeaderChange = useCallback(
+		( newHeaderContent ) => {
+			setAttributes( { headerContent: newHeaderContent } );
+		},
+		[ setAttributes ]
+	);
+
+	return (
+		<>
+			<InspectorControls>
+				<TemplateControls templateName={ templateName } onChange={ handleTemplateChange } />
+				{ templateName === 'custom' && (
+					<ColorControls customColor={ customColor } onColorChange={ handleColorChange } />
+				) }
+				<IconControls
+					customIcon={ customIcon }
+					currentColor={ currentColor }
+					templateName={ templateName }
+					onIconChange={ handleIconChange }
+				/>
+			</InspectorControls>
+
+			<aside
+				className={ `o2-blocks-callout o2-blocks-callout--${ templateName }` }
+				style={ customStyle }
+			>
+				<CalloutHeader
+					content={ headerContent }
+					onChange={ handleHeaderChange }
+					placeholder={ __( 'Header' ) }
+					color={ currentColor }
+					iconName={ currentIcon }
+				/>
+
+				<InnerBlocks
+					template={ [ [ 'core/paragraph', {} ] ] }
+					templateInsertUpdatesSelection={ false }
+				/>
+			</aside>
+		</>
+	);
+};
+
+const Save = ( { attributes } ) => {
+	const {
+		calloutTemplate: templateName = 'custom',
+		customColor = CALLOUT_TEMPLATE_DEFAULT_COLOR,
+		customIcon = '',
+		headerContent = '',
+	} = attributes;
+
+	const currentColor = getCurrentColor( templateName, customColor );
+	const currentIcon = getIcon( templateName, customIcon );
+
+	const customStyle =
+		templateName === 'custom'
+			? { '--custom-color': normalizeColor( customColor ) || CALLOUT_TEMPLATE_DEFAULT_COLOR }
+			: {};
+
+	return (
+		<aside
+			className={ `o2-blocks-callout o2-blocks-callout--${ templateName }` }
+			style={ customStyle }
+		>
+			<h2 className="o2-blocks-callout__header" style={ { color: currentColor } }>
+				<CalloutIcon iconName={ currentIcon } color={ currentColor } />
+				<span className="o2-blocks-callout__header-text">
+					<RichText.Content value={ headerContent } style={ { color: currentColor } } />
+				</span>
+			</h2>
+
+			<InnerBlocks.Content />
+		</aside>
+	);
+};
+
+// === BLOCK REGISTRATION ===
+
+registerBlockType( 'a8c/callout', {
+	title: __( 'Callout' ),
+	description: __( 'Draw attention with a styled callout box.' ),
+	icon: 'megaphone',
+	category: 'a8c',
+	keywords: [ __( 'callout' ), __( 'info' ), __( 'warning' ), __( 'tip' ) ],
+	supports: {
+		align: [ 'wide', 'full' ],
+		html: false,
+	},
+	attributes: {
+		calloutTemplate: {
+			type: 'string',
+			default: 'custom',
+		},
+		customColor: {
+			type: 'string',
+			default: CALLOUT_TEMPLATE_DEFAULT_COLOR,
+		},
+		customIcon: {
+			type: 'string',
+			default: '',
+		},
+		headerContent: {
+			type: 'string',
+			source: 'html',
+			selector: '.o2-blocks-callout__header-text',
+			default: '',
+		},
+	},
+	transforms,
+	edit: Edit,
+	save: Save,
+} );
```

Comment:
it would be helpful to create a `blocks.json` file for this so that if we need information about it from PHP then we can read it.

Reply:
Sounds good! Added in https://github.com/Automattic/wp-calypso/pull/104352/commits/bdbc2aea9c12e310e70a18881464ee39e2063b6c!